ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java

```diff
@@ -30,491 +33,266 @@
-        this.options.addOption(insecure);
-        this.options.addOption(http);
-        this.options.addOption(version);
-        this.options.addOption(verbose);
-        this.options.addOption(debug);
-        this.options.addOption(trace);
-        this.options.addOption(logVerbose);
-        this.options.addOption(logDebug);
-        this.options.addOption(logTrace);
-        this.options.addOption(viewType);
-        this.options.addOption(completed);
-        this.options.addOption(sync);
-        this.options.addOption(numberOfFiles);
-        this.options.addOption(sizeOfFiles);
-        this.options.addOption(bufferSize);
-        this.options.addOption(numberOfThreads);
-        this.options.addOption(ignoreErrors);
-        this.options.addOption(noFollowSymlinks);
-        this.options.addOption(followSymLinks);
-        this.options.addOption(metadata);
-        this.options.addOption(modifyParams);
-        this.options.addOption(filterParams);
-        this.options.addOption(discard);
-        this.options.addOption(ignoreNamingConflicts);
-        this.options.addOption(inCache);
-        this.options.addOption(verifyPercent);
-        this.options.addOption(ejectLabel);
-        this.options.addOption(ejectLocation);
+        // parse all values required to select help, query version, or specify COMMAND, set up the client, and configure logging.
+        addOption(ENDPOINT, "endpoint");
+        addOption(ACCESS_KEY, "accessKeyId");
+        addOption(SECRET_KEY, "secret_key");
+        addOption(COMMAND, "command");
+        addOption(PROXY, "proxy");
+        addOption(HTTP, "http");
+        addOption(INSECURE, "insecure");
+        addOption(PRINT_HELP, "help");
+        addOption(COMMAND_HELP, "command_help");
+        addOption(VERBOSE, "verbose");
+        addOption(DEBUG, "debug");
+        addOption(TRACE, "trace");
+        addOption(LOG_VERBOSE, "log-verbose");
+        addOption(LOG_DEBUG, "log-debug");
+        addOption(LOG_TRACE, "log-trace");
+        addOption(PRINT_VERSION, "version");
+        addOption(VIEW_TYPE, "view_type");
+        addOption(RETRIES, "retries");
+        addOption(BUFFER_SIZE, "buffer_size");
+
         this.processCommandLine();
     }
 
-    private void processCommandLine() throws ParseException, BadArgumentException {
+    private String[] rootArgumentsOnly() throws BadArgumentException {
+        // Present only COMMAND, help and version arguments to parser, After the COMMAND
+        // is instantiated, the full parse will be run against the complete Options list.
+        final List<String> rootArguments = new ArrayList<String>();
+
+        for (int i = 0; i < this.args.length; i++) {
+            final String token = this.args[i];
+            // allow get COMMAND (-c) and subsequent argument
+            // or --output-format and subsequent argument
+            if (matchesOption(COMMAND, token) || matchesOption(VIEW_TYPE,  token) ||
```

Comment:
This would be a lot cleaner if we had a set of options that were 'root' arguments